### ISSUE TITLE
Doesn't see backup

### ISSUE THREAD

**Issue by Iskandeor** [gwatts/pinfinder#37]
Pinfinder doesn't see backup of iPhone 7 iOS 11.4.1. For iPhone 5s iOS 11.4.1 it works. Can you please help me to fix this problem?

**Comment by gwatts** [comment]
That sounds very odd.

Is this on Windows?  If so, which version of Windows and which version of iTunes?

Can you share a screenshot of pinfinder's output (can blank out the passcodes, of course), or email to gareth@omnipotent.net

**Comment by Iskandeor** [comment]
It's on Mac El Capitan 10.11.6, iTunes 12.8.0 and pinfinder is 1.6.1. I has send you screenshot

**Comment by Iskandeor** [comment]
![2018-09-15 12 43 08](https://user-images.githubusercontent.com/43212590/45584960-27d39780-b8e5-11e8-8d06-2b0668b8b317.png)

**Comment by gwatts** [comment]
well that just seems very weird.. if you open a terminal, can you run the following and paste the output?

```
find "$HOME/Library/Application Support/MobileSync/Backup" -name Manifest.plist
```

That should show the actual directories that contain backup files; would be a useful first step to figure out why pinfinder isn't seeing them

**Comment by jtiiii** [comment]
> well that just seems very weird.. if you open a terminal, can you run the following and paste the output?
> 
> ```
> find "$HOME/Library/Application Support/MobileSync/Backup" -name Manifest.plist
> ```
> That should show the actual directories that contain backup files; would be a useful first step to figure out why pinfinder isn't seeing them

I didn't see backup too.
Then I ran that 'find' instruction. It didn't work.
output:
>
> ```
> $sudo find "$HOME/Library/Application Support/MobileSync/Backup" -name Manifest.plist
> find: /Users/peizangpin/Library/Application Support/MobileSync/Backup: Operation not permitted
> ```

**Comment by gwatts** [comment]
If you run pinfinder with sudo, it will be run as root and pinfinder will assume $HOME to be /root and thus not find your backup directory... You should absolutely not need to run pinfinder as root for it to wokr - All that should be required is to grant permission to the Terminal permission to access the whole disk, which you can do in the security preferences - See https://pinfinder.net/mac.html - Once that's done, your normal user account should work fine

If you *do* run it as root, you will have to pass it the path to the actual directory containing the backup (the line output from the find command in this case) to pinfinder directly and it will work then

ie.
```
sudo ~/Downloads/pinfinder /Users/peizangopin/Library/<etc>
```
(not including the `Manifest.plist` part)

**Comment by cantkillemall** [comment]
Hi, just tried to retrieve the passcode from an iPad Air 2, but this is the message I got from Pinfinder:
`Last login: Tue Dec  4 18:17:33 on ttys000
macbook-pro-retina-de-thierry:~ tdilenarda$ /Users/tdilenarda/Downloads/pinfinder ; exit;
PIN Finder 1.7.0
iOS Restrictions Passcode Finder
https://pinfinder.net

Sync Directories: [/Users/tdilenarda/Library/Application Support/MobileSync/Backup]
Scanning backups...
failed to open directory "/Users/tdilenarda/Library/Application Support/MobileSync/Backup": open /Users/tdilenarda/Library/Application Support/MobileSync/Backup: operation not permitted
Usage: pinfinder  [flags] [<path to latest iTunes backup directory>]
  -diag
    	Generate a diagnostic pinfinder-debug.zip file
  -license
    	Display license information
  -nopause
    	Set to true to prevent the program pausing for input on completion
Press Enter to exit
`

**Comment by MrJarnould** [comment]
@cantkillemall See this article. I had the same error message as you.  Giving Terminal "Full Disk Access" permission worked for me.

**Comment by gwatts** [comment]
Details on how to setup full disk access are also on the pinfinder site: https://pinfinder.net/mac.html

**Comment by cantkillemall** [comment]
Thanks @MrJarnould! I didn't connect the dots between the Terminal message and this. Worked swiftly

**Comment by gwatts** [comment]
I just released 1.7.1 too that adds a more helpful message if full disk access needs to be setup

**Comment by MrJarnould** [comment]
> I just released 1.7.1 too that adds a more helpful message if full disk access needs to be setup

I was in the middle of writing a comment requesting that you change the docs to specify "Full Disk Access", thanks!

**Comment by jezmccole** [comment]
Hi - 1.7.1 on Mac OS 10.14.2 - seems Pinfinder is still asking for Full disk access to terminal when I have already given this - and restarted. ...   any other thoughts?:

"Please grant "Full Disk Access" to Terminal to run pinfinder" - already done
